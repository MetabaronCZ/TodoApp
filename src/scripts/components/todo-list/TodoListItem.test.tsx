import React from 'react';

import { fireEvent, render } from '@testing-library/react';
import { describe, expect, it, jest } from '@jest/globals';

import { TodoListItem } from 'components/todo-list/TodoListItem';

import { Todo } from 'models/Todo';
import { withMockedProviders } from 'test/component';
import { icos } from 'modules/icos';

const testData: Todo = {
  id: 'test123',
  title: 'Test todo',
  description: '',
  created: 0,
  folder: null,
  isDone: true,
};

describe('components/todo-list/TodoListItem', () => {
  it('should render correctly', () => {
    const tree = render(
      withMockedProviders(
        <TodoListItem
          item={testData}
          onSelect={() => null}
          onDelete={() => null}
        />,
      ),
    );
    expect(tree.container).toMatchSnapshot();
    expect(tree.getByText('Test todo')).toBeInTheDocument();
    expect(tree.getByText('1. 1. 1970')).toBeInTheDocument();
    expect(tree.getByText(icos.success)).toBeInTheDocument();
    expect(tree.getByRole('checkbox')).toBeInTheDocument();
    expect(tree.getByRole('checkbox')).not.toHaveAttribute('checked');
  });

  it('should be selectable', () => {
    const onSelect = jest.fn();

    const tree = render(
      withMockedProviders(
        <TodoListItem
          item={testData}
          selected
          onSelect={onSelect}
          onDelete={() => null}
        />,
      ),
    );
    expect(tree.container).toMatchSnapshot();
    expect(onSelect).toBeCalledTimes(0);

    const checkbox = tree.getByRole('checkbox');
    expect(checkbox).toBeInTheDocument();
    expect(checkbox).toHaveAttribute('checked');

    fireEvent.click(checkbox);
    expect(onSelect).toBeCalledTimes(1);
    expect(onSelect).toBeCalledWith(false);
  });

  it('should be delete-able', () => {
    const onDelete = jest.fn();

    const tree = render(
      withMockedProviders(
        <TodoListItem
          item={testData}
          onSelect={() => null}
          onDelete={onDelete}
        />,
      ),
    );
    expect(onDelete).toBeCalledTimes(0);

    const button = tree.getByRole('button');
    expect(button).toBeInTheDocument();

    fireEvent.click(button);
    expect(onDelete).toBeCalledTimes(1);
  });
});
