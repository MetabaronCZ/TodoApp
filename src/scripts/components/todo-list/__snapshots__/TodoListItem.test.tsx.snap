// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/todo-list/TodoListItem > should be selectable 1`] = `
.c7 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: block;
  width: 20px;
  height: 20px;
  line-height: 20px;
  font-style: normal;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c10 {
  display: block;
  width: 100%;
  border: none;
  outline: none;
  cursor: pointer;
}

.c11 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: auto;
  padding: 5px 10px;
  padding-left: 5px;
  background: #f0f0f0;
}

.c11:hover:not(:disabled) {
  background-color: #e0e0e0;
}

.c11:focus {
  outline: 1px dotted #222;
  outline-offset: -1px;
}

.c11:disabled {
  color: #c0c0c0;
}

.c8 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: auto;
  padding: 5px 10px;
  padding-left: 5px;
  background: #f0f0f0;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8:hover:not(:disabled) {
  background-color: #e0e0e0;
}

.c8:focus {
  outline: 1px dotted #222;
  outline-offset: -1px;
}

.c8:disabled {
  color: #c0c0c0;
}

.c9 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
  height: 30px;
}

.c2 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 1;
  width: 20px;
  height: 20px;
  border: 2px solid #222;
  color: #222;
  background: #fff;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
}

.c2:hover {
  background-color: #fcfcfc;
}

.c2:focus-within {
  outline: 1px dotted #222;
}

.c3 {
  position: absolute;
  top: 0;
  left: 0;
  width: 0;
  height: 0;
  opacity: 0;
  outline: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
}

.c4 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 5px 0;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  width: 20px;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c5 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  white-space: nowrap;
}

<div>
  <div
    class="c0"
  >
    <div
      class=""
    >
      <div
        class="c1"
      >
        <label
          class="c2"
          for="checkbox-1"
        >
          ✔
          <input
            checked=""
            class="c3"
            id="checkbox-1"
            type="checkbox"
          />
        </label>
      </div>
    </div>
    <a
      class="c4"
      href="/todo/test123"
    >
      Test todo
    </a>
    <div
      class="c5"
    >
      1. 1. 1970
    </div>
    <div
      class="c6"
      title="Hotovo"
    >
      <i
        class="c7"
      >
        ✔
      </i>
    </div>
    <div
      class=""
    >
      <a
        class="c8"
        href="/todo/test123"
      >
        <i
          class="c7"
        >
          ✎
        </i>
        <span
          class="c9"
        >
          Upravit
        </span>
      </a>
    </div>
    <div
      class=""
    >
      <button
        class="c10 c11"
        type="button"
      >
        <i
          class="c7"
        >
          ✖
        </i>
        <span
          class="c9"
        >
          Smazat
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`components/todo-list/TodoListItem > should render correctly 1`] = `
.c7 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: block;
  width: 20px;
  height: 20px;
  line-height: 20px;
  font-style: normal;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c10 {
  display: block;
  width: 100%;
  border: none;
  outline: none;
  cursor: pointer;
}

.c11 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: auto;
  padding: 5px 10px;
  padding-left: 5px;
  background: #f0f0f0;
}

.c11:hover:not(:disabled) {
  background-color: #e0e0e0;
}

.c11:focus {
  outline: 1px dotted #222;
  outline-offset: -1px;
}

.c11:disabled {
  color: #c0c0c0;
}

.c8 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: auto;
  padding: 5px 10px;
  padding-left: 5px;
  background: #f0f0f0;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8:hover:not(:disabled) {
  background-color: #e0e0e0;
}

.c8:focus {
  outline: 1px dotted #222;
  outline-offset: -1px;
}

.c8:disabled {
  color: #c0c0c0;
}

.c9 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
  height: 30px;
}

.c2 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 1;
  width: 20px;
  height: 20px;
  border: 2px solid #222;
  color: #222;
  background: #fff;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
}

.c2:hover {
  background-color: #fcfcfc;
}

.c2:focus-within {
  outline: 1px dotted #222;
}

.c3 {
  position: absolute;
  top: 0;
  left: 0;
  width: 0;
  height: 0;
  opacity: 0;
  outline: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
}

.c4 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 5px 0;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  width: 20px;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c5 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  white-space: nowrap;
}

<div>
  <div
    class="c0"
  >
    <div
      class=""
    >
      <div
        class="c1"
      >
        <label
          class="c2"
          for="checkbox-0"
        >
          <input
            class="c3"
            id="checkbox-0"
            type="checkbox"
          />
        </label>
      </div>
    </div>
    <a
      class="c4"
      href="/todo/test123"
    >
      Test todo
    </a>
    <div
      class="c5"
    >
      1. 1. 1970
    </div>
    <div
      class="c6"
      title="Hotovo"
    >
      <i
        class="c7"
      >
        ✔
      </i>
    </div>
    <div
      class=""
    >
      <a
        class="c8"
        href="/todo/test123"
      >
        <i
          class="c7"
        >
          ✎
        </i>
        <span
          class="c9"
        >
          Upravit
        </span>
      </a>
    </div>
    <div
      class=""
    >
      <button
        class="c10 c11"
        type="button"
      >
        <i
          class="c7"
        >
          ✖
        </i>
        <span
          class="c9"
        >
          Smazat
        </span>
      </button>
    </div>
  </div>
</div>
`;
